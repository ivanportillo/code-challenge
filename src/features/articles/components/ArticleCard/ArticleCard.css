.ArticleCard {
    border: 1px solid #BDBDBD;
    border-radius: 4px;
    padding: 2em;
}

.Title {
    font-size: 1.5em;
    font-weight: bold;
}

.Author {
    padding-bottom: 10px;
    color: #757575;
}

.Excerpt {

}