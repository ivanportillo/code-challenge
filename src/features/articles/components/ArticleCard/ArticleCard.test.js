import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import ArticleCard from './ArticleCard';

const fakeData = {
  title: 'Article title',
  author: 'Article author',
  excerpt: 'Article excerpt',
};

describe('Article Card', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <ArticleCard
        title={fakeData.title}
        author={fakeData.author}
        excerpt={fakeData.excerpt}
      />, div);
  });

  it('should render an article', () => {
    const article = shallow(
      <ArticleCard
        title={fakeData.title}
        author={fakeData.author}
        excerpt={fakeData.excerpt}
      />);
    expect(article).toMatchSnapshot();
  });
});
