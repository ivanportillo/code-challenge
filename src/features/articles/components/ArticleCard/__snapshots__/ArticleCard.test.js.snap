exports[`Article Card should render an article 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="ArticleCard">
    <div
        className="Title">
        Article title
    </div>
    <div
        className="Author">
        Article author
    </div>
    <div
        className="Excerpt">
        Article excerpt
    </div>
</div>,
  "nodes": Array [
    <div
      className="ArticleCard">
      <div
            className="Title">
            Article title
      </div>
      <div
            className="Author">
            Article author
      </div>
      <div
            className="Excerpt">
            Article excerpt
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <ArticleCard
        author="Article author"
        excerpt="Article excerpt"
        title="Article title" />,
      "_debugID": 6,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "author": "Article author",
          "excerpt": "Article excerpt",
          "title": "Article title",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="ArticleCard">
          <div
                    className="Title">
                    Article title
          </div>
          <div
                    className="Author">
                    Article author
          </div>
          <div
                    className="Excerpt">
                    Article excerpt
          </div>
</div>,
        "_debugID": 7,
        "_renderedOutput": <div
          className="ArticleCard">
          <div
                    className="Title">
                    Article title
          </div>
          <div
                    className="Author">
                    Article author
          </div>
          <div
                    className="Excerpt">
                    Article excerpt
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ArticleCard
    author="Article author"
    excerpt="Article excerpt"
    title="Article title" />,
}
`;
